contributing.md

```diff
@@ -0,0 +1,43 @@
+# Contributing
+
+Great that you're willing to contribute to Pylake!
+
+This document is intended to provide you with some instructions on how to contribute to Pylake.
+
+The first step in contributing to Pylake is to open a GitHub issue. By first discussing the improvements you wish to make, we aim to prevent unnecessary work up front. We can also help by making suggestions on where said functionality might best fit in.
+
+If you feel uncomfortable immediately committing code, you could also consider other ways to contribute. Think of improvements to the documentation, additional tutorials or other tasks that generally improve Pylake and benefit its users. If you have an idea for a new feature, but do not know how to start implementing it, consider opening an issue for it. The same applies to bug fixes.
+
+## Submitting a bug report
+
+If you encounter a bug, please verify that the bug still exists on the latest version of Pylake. You can find instructions on how to update Pylake [here](https://lumicks-pylake.readthedocs.io/en/stable/install.html#updating).
+
+If the bug persists, please open an issue describing the bug. This issue should at least contain the following information:
+
+- A short summary of the bug. In most cases, this should only be a couple of sentences.
+- A code snippet to reproduce the bug. Please do your best to reduce the code snippet to the minimum required.
+- The result of the code snippet.
+- The expected result of the code snippet (which is different from the obtained result because of the bug).
+
+Please also make sure you mention the version of Python, the platform and the version of Pylake. If reproducing the bug relies on data, please make sure this data is accessible to us.
+
+## Preparing a Pull Request (PR)
+
+Code and documentation contributions will generally proceed through a Pull Request.
+
+- Please provide a clear PR description detailing the change in functionality the PR provides. If the implemented method is based on a paper, please include a reference to the paper.
+- If the PR is related to an open issue, please refer to this issue in the PR description.
+- Try and keep the pull request manageable in terms of size. If a new feature is complex and requires multiple steps, consider breaking it up into multiple Pull Requests for the different sub-problems. If you struggle with this, please open an issue beforehand, and we can try and help you.
+- Keep a clean commit history. This will greatly aid in getting your PR reviewed quickly.
```

Comment:
Shorter version of above two bullet points:

-  Keep the pull request manageable in terms of size. If a new feature is complex and requires multiple steps, consider breaking it up into multiple Pull Requests for the different sub-problems. 
- Keep a clean commit history.